### PR TITLE
:fix Do not compute device fingerprint when security image is off

### PR DESCRIPTION
**Pull request by magizh-okta**
* When user tabs out after typing username, we should not compute device fingerprint
if sec image feature is off.

OKTA-218737

**BEfore** 
Screen Shot 2019-04-02 at 7 47 06 AM

**After** 
Screen Shot 2019-04-02 at 7 49 58 AM

secdev

### CODE REVIEW

File: test/unit/spec/IDPDiscovery_spec.js [hunk 1]
```diff
@@ -594,6 +594,25 @@ function (Q, OktaAuth, WidgetUtil, Okta, Util, AuthContainer, IDPDiscoveryForm,
     });
 
     Expect.describe('Device Fingerprint', function () {
+      itp(`does not compute device fingerprint if security image is off, deviceFingerprinting
+        is true and useDeviceFingerprintForSecurityImage is true)`, function () {
+        spyOn(DeviceFingerprint, 'generateDeviceFingerprint').and.callFake(function () {
```
Comment (gowthamidommety-okta) [inline test/unit/spec/IDPDiscovery_spec.js]: Should this just be spyOn? We don't need to define the fake function because we are not calling it right?

File: test/unit/spec/PrimaryAuth_spec.js [hunk 2]
```diff
@@ -985,15 +985,33 @@ function (Q, OktaAuth, LoginUtil, Okta, Util, AuthContainer, PrimaryAuthForm, Be
...
       itp(`contains fingerprint header in get security image request if deviceFingerprinting
         is true (useDeviceFingerprintForSecurityImage defaults to true)`, function () {
         spyOn(DeviceFingerprint, 'generateDeviceFingerprint').and.callFake(function () {
           var deferred = Q.defer();
           deferred.resolve('thisIsTheDeviceFingerprint');
           return deferred.promise;
         });
-        return setup({ features: { securityImage: true, deviceFingerprinting: true,
-          useDeviceFingerprintForSecurityImage: true }})
+        return setup({ features: { securityImage: true, deviceFingerprinting: true }})
```
Comment (gowthamidommety-okta) [inline test/unit/spec/PrimaryAuth_spec.js]: Is useDeviceFingerprintForSecurityImage true here?

Comment (magizh-okta) [reply test/unit/spec/PrimaryAuth_spec.js]: Yeah it is true by default.(which is what we are testing here) 

```
contains fingerprint header in get security image request if deviceFingerprinting
        is true (useDeviceFingerprintForSecurityImage defaults to true)
```

File: test/unit/spec/IDPDiscovery_spec.js [hunk 3]
```diff
@@ -594,6 +594,21 @@ function (Q, OktaAuth, WidgetUtil, Okta, Util, AuthContainer, IDPDiscoveryForm,
     });
 
     Expect.describe('Device Fingerprint', function () {
+      itp(`does not compute device fingerprint if security image is off, deviceFingerprinting
+        is true and useDeviceFingerprintForSecurityImage is true)`, function () {
```
Comment (mauriciocastillosilva-okta) [inline test/unit/spec/IDPDiscovery_spec.js]: seems that there is an extra parenthesis in the description... maybe we can shorten this to `is not computed if securityImage is off, deviceFingerprinting is true and useDeviceFingerprintForSecurityImage is true`?

File: test/unit/spec/PrimaryAuth_spec.js [hunk 4]
```diff
@@ -985,15 +985,29 @@ function (Q, OktaAuth, LoginUtil, Okta, Util, AuthContainer, PrimaryAuthForm, Be
     });
 
     Expect.describe('Device Fingerprint', function () {
+      itp(`does not compute device fingerprint if security image is off, deviceFingerprinting
+        is true and useDeviceFingerprintForSecurityImage is true)`, function () {
```
Comment (mauriciocastillosilva-okta) [inline test/unit/spec/PrimaryAuth_spec.js]: nit: same here about the extra parenthesis that is not needed
